### ISSUE TITLE
poetry resolver seems to ignore sys_platform marker

### ISSUE THREAD

**Issue by kadrach** [sdispater/poetry#490]
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Ubuntu Bionic
- **Poetry version**: Poetry 0.12.0a3
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: https://gist.github.com/kadrach/24e29714f9cdb5428c60ff9afb9884a9

## Issue
Running on Linux. The Poetry resolver flags `docker ^3.5` as depending on pypiwin32. The [setup.py](https://github.com/docker/docker-py/blob/3.5.0/setup.py) appears to include correct environment markers. The PyPi API https://pypi.org/pypi/docker/json does too.

```
   requires_dist: [
     ...
     pypiwin32 (==223); sys_platform == \"win32\" and python_version >= \"3.6\",
     pypiwin32 (==219); sys_platform == \"win32\" and python_version < \"3.6\",
   ]
```

**Comment by kadrach** [comment]
Just occured to me - this may not be an issue in the resolver, given this package is required on windows. But when I do `poetry install -vvv`:

```
% poetry install -vvv
Creating virtualenv test-foo-py3.6 in /home/neckerskorn/.cache/pypoetry/virtualenvs
Using virtualenv: /home/neckerskorn/.cache/pypoetry/virtualenvs/test-foo-py3.6
Installing dependencies from lock file


Package operations: 11 installs, 0 updates, 0 removals

  - Installing certifi (2018.8.24)
  - Installing chardet (3.0.4)
  - Installing idna (2.7)
  - Installing pywin32 (224)

[EnvCommandError]
Command ['/home/neckerskorn/.cache/pypoetry/virtualenvs/test-foo-py3.6/bin/pip', 'install', '--no-dep
s', 'pywin32==224'] errored with the following output:
Collecting pywin32==224
  Could not find a version that satisfies the requirement pywin32==224 (from versions: )
No matching distribution found for pywin32==224
You are using pip version 9.0.1, however version 18.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.

Exception trace:
 /home/neckerskorn/.poetry/lib/poetry/_vendor/py3.6/cleo/application.py in run() at line 94
   status_code = self.do_run(input_, output_)
 /home/neckerskorn/.poetry/lib/poetry/console/application.py in do_run() at line 88
   return super(Application, self).do_run(i, o)
 /home/neckerskorn/.poetry/lib/poetry/_vendor/py3.6/cleo/application.py in do_run() at line 197
   status_code = command.run(input_, output_)
 /home/neckerskorn/.poetry/lib/poetry/console/commands/command.py in run() at line 77
   return super(BaseCommand, self).run(i, o)
 /home/neckerskorn/.poetry/lib/poetry/_vendor/py3.6/cleo/commands/base_command.py in run() at line 146
   status_code = self.execute(input_, output_)
 /home/neckerskorn/.poetry/lib/poetry/_vendor/py3.6/cleo/commands/command.py in execute() at line 107
   return self.handle()
 /home/neckerskorn/.poetry/lib/poetry/console/commands/install.py in handle() at line 50
   return installer.run()
 /home/neckerskorn/.poetry/lib/poetry/installation/installer.py in run() at line 76
   self._do_install(local_repo)
 /home/neckerskorn/.poetry/lib/poetry/installation/installer.py in _do_install() at line 286
   self._execute(op)
 /home/neckerskorn/.poetry/lib/poetry/installation/installer.py in _execute() at line 294
   getattr(self, "_execute_{}".format(method))(operation)
 /home/neckerskorn/.poetry/lib/poetry/installation/installer.py in _execute_install() at line 319
   self._installer.install(operation.package)
 /home/neckerskorn/.poetry/lib/poetry/installation/pip_installer.py in install() at line 86
   self.run(*args)
 /home/neckerskorn/.poetry/lib/poetry/installation/pip_installer.py in run() at line 101
   return self._env.run("pip", *args, **kwargs)
 /home/neckerskorn/.poetry/lib/poetry/utils/env.py in run() at line 437
   return super(VirtualEnv, self).run(bin, *args, **kwargs)
 /home/neckerskorn/.poetry/lib/poetry/utils/env.py in run() at line 257
   raise EnvCommandError(e)

install [--no-dev] [--dry-run] [-E|--extras EXTRAS] [--develop DEVELOP]
```

**Comment by sdispater** [comment]
There actually was an issue in the resolver regarding markers. This has already been fixed on the `develop` branch (commit 11c62cb8830fcdea8163b3d2c4ae9e4d9384602b) and will be available in the next prerelease.

**Comment by kadrach** [comment]
Awesome, thank you for the quick response @sdispater. Let me figure out how to install poetry from source and I'll report back.

**Comment by kadrach** [comment]
Can confirmed this is resolved with poetry built from develop (1af7bb3).
